### PR TITLE
refactor chat package to use its own G, and remove chat types off of libkb.G CORE-4984

### PR DESCRIPTION
**Pull request by mmaxim**
The patch does the following:

1.) Move most of `chat_local.go` into the `chat` package under the new name `server.go`. 
2.) Remove all chat types off of `libkb.G`.
3.) Create a new form of global context in the `chat/globals` package called `Context`, which is a composition of `libkb.G` and `globals.ChatContext`. 
4.) `globals.ChatContext` has all the global chat objects on it.
5.) Move a lot of stuff around so things keep working. 

Hope this is clean enough, let me know if you think there is a better way.

### CODE REVIEW

**Comment by mmaxim** [comment]
Oh I also dropped the dependency of Saltpack V2 on chat, and just copied the (short) function to get TLF crypt keys into that world.

File: go/service/main.go [hunk 1]
```diff
@@ -325,7 +335,7 @@ func (d *Service) startupGregor() {
 		// Create gregorHandler instance first so any handlers can connect
 		// to it before we actually connect to gregor (either gregor is down
 		// or we aren't logged in)
-		d.gregor = newGregorHandler(d.G())
+		d.gregor = newGregorHandler(globals.NewContext(d.G(), d.ChatG()))
```
Comment (patrickxb) [inline go/service/main.go]: doesn't `newGregorHandler(d.ChatG())` suffice?

Comment (mmaxim) [inline go/service/main.go]: `ChatG` just returns `globals.ChatContext`, not `globals.Context` (that might be too confusing). `globals.ChatContext` does not have `libkb.G` stuff on it, and `gregorHandler` needs that.

Comment (patrickxb) [inline go/service/main.go]: ok, I think the `ChatG` name confused me.

**Pull request merged**